### PR TITLE
more refactoring around `Felt`

### PR DESCRIPTION
**Pull request by tcoratger**
# Pull Request type

Time spent on this PR:

Resolves: #{Issue number}

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by eugypalu — APPROVED** [review]

**Review by greged93 — COMMENTED** [review]
I'm not too much into replacing `Felt::from` into `into()`, I don't think it brings much and removes some readability in the code

File: crates/ef-testing/src/evm_sequencer/evm_state/v0.rs [hunk 1]
```diff
@@ -269,7 +269,7 @@ impl Evm for KakarotSequencer {
         let chain_id = self.chain_id();
         self.execute(
             starknet_transaction
-                .try_into_execution_transaction(Felt::from(chain_id))
+                .try_into_execution_transaction(chain_id.into())
```
Comment (tcoratger) [inline crates/ef-testing/src/evm_sequencer/evm_state/v0.rs:272]:
```suggestion
                .try_into_execution_transaction(Felt::from(chain_id))
```

File: crates/ef-testing/src/evm_sequencer/evm_state/v1.rs [hunk 2]
```diff
@@ -224,7 +224,7 @@ impl Evm for KakarotSequencer {
             let index = chunk_index / 256;
             let offset = chunk_index % 256;
             let storage_pointer =
-                inner_byte_array_pointer(*bytecode_base_address.0.key(), Felt::from(index));
+                inner_byte_array_pointer(*bytecode_base_address.0.key(), index.into());
```
Comment (tcoratger) [inline crates/ef-testing/src/evm_sequencer/evm_state/v1.rs:227]:
```suggestion
                inner_byte_array_pointer(*bytecode_base_address.0.key(), Felt::from(index));
```

File: crates/ef-testing/src/evm_sequencer/evm_state/v1.rs [hunk 3]
```diff
@@ -299,7 +299,7 @@ impl Evm for KakarotSequencer {
         let chain_id = self.chain_id();
         self.execute(
             starknet_transaction
-                .try_into_execution_transaction(Felt::from(chain_id))
+                .try_into_execution_transaction(chain_id.into())
```
Comment (tcoratger) [inline crates/ef-testing/src/evm_sequencer/evm_state/v1.rs:302]:
```suggestion
                .try_into_execution_transaction(Felt::from(chain_id))
```

File: crates/ef-testing/src/evm_sequencer/sequencer/v0.rs [hunk 4]
```diff
@@ -30,7 +29,7 @@ lazy_static! {
             (KAKAROT_NATIVE_TOKEN_ADDRESS, *ETH_FEE_TOKEN_ADDRESS.0.key()),
             (KAKAROT_ACCOUNT_CONTRACT_CLASS_HASH, ACCOUNT_CONTRACT_CLASS_HASH.0),
             (KAKAROT_CAIRO1_HELPERS_CLASS_HASH, CAIRO1_HELPERS_CLASS_HASH.0),
-            (KAKAROT_BLOCK_GAS_LIMIT, Felt::from(BLOCK_GAS_LIMIT)),
+            (KAKAROT_BLOCK_GAS_LIMIT, BLOCK_GAS_LIMIT.into()),
```
Comment (tcoratger) [inline crates/ef-testing/src/evm_sequencer/sequencer/v0.rs:32]:
```suggestion
            (KAKAROT_BLOCK_GAS_LIMIT, Felt::from(BLOCK_GAS_LIMIT)),
```

File: crates/ef-testing/src/evm_sequencer/sequencer/v1.rs [hunk 5]
```diff
@@ -26,7 +25,7 @@ lazy_static! {
             (OWNABLE_OWNER, *KAKAROT_OWNER_ADDRESS.0.key()),
             (KAKAROT_NATIVE_TOKEN_ADDRESS, *ETH_FEE_TOKEN_ADDRESS.0.key()),
             (KAKAROT_ACCOUNT_CONTRACT_CLASS_HASH, ACCOUNT_CONTRACT_CLASS_HASH.0),
-            (KAKAROT_BLOCK_GAS_LIMIT, Felt::from(BLOCK_GAS_LIMIT)),
+            (KAKAROT_BLOCK_GAS_LIMIT, BLOCK_GAS_LIMIT.into()),
```
Comment (tcoratger) [inline crates/ef-testing/src/evm_sequencer/sequencer/v1.rs:28]:
```suggestion
            (KAKAROT_BLOCK_GAS_LIMIT, Felt::from(BLOCK_GAS_LIMIT)),
```

File: crates/ef-testing/src/evm_sequencer/utils.rs [hunk 6]
```diff
@@ -52,18 +50,15 @@ pub fn to_broadcasted_starknet_transaction(
...
+            std::iter::once((bytes.len()).into())
+                .chain(pack_byte_array_to_starkfelt_array(
+                    &bytes.into_iter().collect::<Vec<u8>>(),
+                ))
                 .collect()
         }
         #[cfg(not(feature = "v0"))]
         {
-            bytes.into_iter().map(Felt::from).collect()
+            bytes.into_iter().map(Into::into).collect()
```
Comment (tcoratger) [inline crates/ef-testing/src/evm_sequencer/utils.rs:61]:
```suggestion
            bytes.into_iter().map(Felt::from).collect()
```

File: crates/sequencer/src/state.rs [hunk 7]
```diff
@@ -82,7 +82,7 @@ impl BlockifierState for &mut State {
             .copied()
             .unwrap_or_default();
 
-        if current_nonce == Nonce(Felt::from(u64::MAX)) {
+        if current_nonce == Nonce(u64::MAX.into()) {
```
Comment (tcoratger) [inline crates/sequencer/src/state.rs:85]:
```suggestion
        if current_nonce == Nonce(Felt::from(u64::MAX)) {
```

**Comment by tcoratger** [comment]
> I'm not too much into replacing `Felt::from` into `into()`, I don't think it brings much and removes some readability in the code

@greged93 In fact my objective was to depend as little as possible on the word `Felt` because it can change, etc. depending on updates and with `into` the inference is automatic. But you're right, it limits readability.

So I reverted the changes to keep this readability but I kept the main refactor that I wanted to carry out at the start of this PR if you agree with this one.

**Review by greged93 — APPROVED** [review]

**Pull request merged**